### PR TITLE
Lock dialog form fields while processing

### LINKED ISSUE #31654: Link share dialog must block during creation
### Steps

1. Enable encryption (this is to cause longer delays during share creation)
1. Create a big folder with lots of files
1. Share said folder with link
1. In the dialog, click "Save"

### Expected result
During saving, which takes longer, the dialog must be disabled and a spinner must be displayed to tell me it's working.

### Actual result
No spinner appears, I can still click inside the dialog fields as if I didn't properly click "Save".
If I click "Save" again, two links are created.

### Version
ownCloud 10.0.8

### PR DESCRIPTION
**Pull request by lordelix**
## Description
Lock dialog form fields while processing to prevent further tinkering with its contents.

## Related Issue

Fixes https://github.com/owncloud/core/issues/31654

## How Has This Been Tested?

- Visual tests on IE, FF and Chrome
- Added unit tests

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: core/js/tests/specs/sharedialoglinkshareviewSpec.js [hunk 1]
```diff
@@ -257,6 +257,10 @@ describe('OC.Share.ShareDialogLinkShareView', function() {
 			}
 		});
 
+		it('locks all input fields while processing', function() {
+			view._save();
+			expect(view.$('input, textarea, select, button').prop('disabled')).toEqual(true);
```
Comment (PVince81) [inline core/js/tests/specs/sharedialoglinkshareviewSpec.js]: also test that they are unlocked again if an error comes up

**Review by PVince81 — APPROVED** [review]
:+1:

**Comment by PVince81** [comment]
@felixheidecke please backport

**Comment by phil-davis** [comment]
Backport ``stable10`` #31928

**Pull request merged**